x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/index.test.tsx

```diff
@@ -0,0 +1,629 @@
+// unified-field-list is is reporiting multiple analytics events
+jest.mock(`@kbn/analytics-client`);
+
+const columnsToDisplay = [
+  ...defaultUdtHeaders,
+  {
+    columnHeaderType: 'not-filtered' as ColumnHeaderType,
+    id: 'event.severity',
+  },
+];
+
+const TestComponent = (props: Partial<ComponentProps<typeof UnifiedTimeline>>) => {
+  const testComponentDefaultProps: ComponentProps<typeof QueryTabContent> = {
+    columns: getColumnHeaders(columnsToDisplay, mockSourcererScope.browserFields),
+    activeTab: TimelineTabs.query,
+    rowRenderers: [],
+    timelineId: TimelineId.test,
+    itemsPerPage: 10,
+    itemsPerPageOptions: [],
+    sort: [
+      {
+        columnId: '@timestamp',
+        columnType: 'date',
+        esTypes: ['date'],
+        sortDirection: 'desc',
+      },
+    ],
+    events: structuredClone(mockTimelineData),
+    refetch: jest.fn(),
+    totalCount: mockTimelineData.length,
+    onEventClosed: jest.fn(),
+    expandedDetail: {},
+    showExpandedDetails: false,
+    onChangePage: jest.fn(),
+    dataLoadingState: DataLoadingState.loaded,
+    updatedAt: Date.now(),
+    isTextBasedQuery: false,
+  };
+
+  const dispatch = useDispatch();
+
+  // populating timeline so that it is not blank
+  useEffect(() => {
+    dispatch(
+      timelineActions.applyKqlFilterQuery({
+        id: TimelineId.test,
+        filterQuery: {
+          kuery: {
+            kind: 'kuery',
+            expression: '*',
+          },
+          serializedQuery: '*',
+        },
+      })
+    );
+  }, [dispatch]);
+
+  return <UnifiedTimeline {...testComponentDefaultProps} {...props} />;
+};
+
+const customStore = createMockStore();
+
+const TestProviderWrapperWithCustomStore: React.FC = ({ children }) => {
+  return <TestProviders store={customStore}>{children}</TestProviders>;
+};
+
+const renderTestComponents = (props?: Partial<ComponentProps<typeof TestComponent>>) => {
+  return render(<TestComponent {...props} />, {
+    wrapper: TestProviderWrapperWithCustomStore,
+  });
+};
+
+const loadPageMock = jest.fn();
+
+const useTimelineEventsMock = jest.fn(() => [
+  false,
+  {
+    events: cloneDeep(mockTimelineData),
+    pageInfo: {
+      activePage: 0,
+      totalPages: 10,
+    },
+    refreshedAt: Date.now(),
+    totalCount: 70,
+    loadPage: loadPageMock,
+  },
+]);
+
+const useSourcererDataViewMocked = jest.fn().mockReturnValue({
+  ...mockSourcererScope,
+});
+
+const { storage: storageMock } = createSecuritySolutionStorageMock();
+const mockTimelineFilterManager = createFilterManagerMock();
+
+describe('unified timeline', () => {
+  const kibanaServiceMock: StartServices = {
+    ...createStartServicesMock(),
+    storage: storageMock,
+    timelineFilterManager: mockTimelineFilterManager,
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    storageMock.clear();
+    cleanup();
+    localStorage.clear();
+  });
+
+  beforeEach(() => {
+    const ONE_SECOND = 1000;
+    jest.setTimeout(10 * ONE_SECOND);
```

Comment:
What is the timeout for? Mind adding a comment? Is it for the resize to take place?

Reply:
Done. Added the comment. Since tests are rendering complete table with added data, they are taking some time.